src/solver/conv_hip_implicit_gemm_bwd_data_xdlops.cpp

```diff
@@ -0,0 +1,391 @@
+{
+    if(total_size == -1)
+        this->HeuristicInit(problem);
+    assert(total_size != -1);
+    if((index + 1) < total_size)
+    {
+        ++index;
+        return true;
+    }
+    else
+        return false;
+}
+
+bool PerformanceConfigHipImplicitGemmBwdXdlops::IsValidValue() const { return index < total_size; }
+
+bool PerformanceConfigHipImplicitGemmBwdXdlops::IsValid(const ProblemDescription& problem) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem;
+    return false;
+#else
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenHalf: return CheckIsSupportCKArgs<ck::half_t>(problem);
+    case miopenFloat: return CheckIsSupportCKArgs<float>(problem);
+    case miopenInt8:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+bool PerformanceConfigHipImplicitGemmBwdXdlops::operator==(
+    const PerformanceConfigHipImplicitGemmBwdXdlops& other) const
+{
+    return this->index == other.index;
+}
+
+PerformanceConfigHipImplicitGemmBwdXdlops
+ConvHipImplicitGemmBwdXdlops::GetDefaultPerformanceConfig(const ProblemDescription& problem) const
+{
+    PerformanceConfigHipImplicitGemmBwdXdlops pp;
+    pp.HeuristicInit(problem);
+    return pp;
+}
+
+bool ConvHipImplicitGemmBwdXdlops::IsValidPerformanceConfig(
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmBwdXdlops& config) const
+{
+    return config.IsValid(problem);
+}
+
+PerformanceConfigHipImplicitGemmBwdXdlops
+ConvHipImplicitGemmBwdXdlops::Search(const ConvolutionContext& ctx,
+                                     const ProblemDescription& problem,
+                                     const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, problem, invoke_ctx);
+}
+
+bool ConvHipImplicitGemmBwdXdlops::IsApplicable(const ConvolutionContext& ctx,
+                                                const ProblemDescription& problem) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = problem;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_BWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(problem.conv_problem.GetInDataType() != problem.conv_problem.GetWeightsDataType() ||
+       problem.conv_problem.GetWeightsDataType() != problem.conv_problem.GetOutDataType() ||
+       problem.conv_problem.GetInDataType() != problem.conv_problem.GetOutDataType())
+        return false;
+    if(!problem.direction.IsBackwardData())
+        return false;
+    if(!problem.Is2d())
+        return false;
+    if(!problem.IsLayoutNHWC())
+        return false;
+    const std::string& arch = ctx.GetStream().GetDeviceName();
+    if(arch != "gfx908" && arch != "gfx90a")
```

Comment:
What about `gfx1030`?

Reply:
No, currently CK does not support navi.

Reply:
May be miopen log saying CK does not support this `arch`?

Reply:
Not sure, it might be overkill, since we have to do the same for all the other solvers's `IsApplicable`.